textile-utils-api-v1/src/main/java/net/textilemc/textile/utils/mixin/BlockCacherMixin.java

```diff
@@ -0,0 +1,123 @@
+package net.textilemc.textile.utils.mixin;
+
+import net.minecraft.block.Block;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.item.Item;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.util.Util;
+import net.textilemc.textile.utils.Block.CustomBlock;
+import net.textilemc.textile.utils.util.Identifier;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+
+@Mixin(MinecraftClient.class)
+public abstract class BlockCacherMixin {
+
+	@Shadow
+	private static File mcDir;
+
+	@Unique
+	private String name;
+
+	//Before you ask, yes, caching is necessary in this case. It fails otherwise complaining about "World cannot be cast to String"
+	@Inject(at=@At("HEAD"), method = "joinWorld(Ljava/lang/String;)V")
+	public void cacheName(String name, CallbackInfo info) {
+		this.name = name;
+	}
+
+	@Inject(at=@At(value = "INVOKE", target = "Lnet/minecraft/client/MinecraftClient;joinWorld(Lnet/minecraft/world/World;Ljava/lang/String;)V", ordinal = 1), method = "joinWorld(Ljava/lang/String;)V")
+	public void removeConflictOlderVersion(CallbackInfo info) {
+		mapBlockToDat(new File(new File(mcDir, "saves"), name));
+
+	}
+	@Inject(at=@At(value = "INVOKE", target = "Lnet/minecraft/client/MinecraftClient;joinWorld(Lnet/minecraft/world/World;Ljava/lang/String;)V", ordinal = 2), method = "joinWorld(Ljava/lang/String;)V")
+	public void removeConflictOlderVersion2(CallbackInfo info) {
+		mapBlockToDat(new File(new File(mcDir, "saves"), name));
+	}
+
+	@Unique
+	private void mapBlockToDat(File saveDir) {
+		if (!(new File(saveDir, "block_registry.dat")).exists()) {
+			System.out.println("Saving Block Registry to .minecraft/saves/" + name + "/block_registry.dat");
```

Comment:
use LOGGER.debug here

Reply:
If `LOGGER` doesn't exist (which it probably doesn't), just create a `@Unique private static final Logger LOGGER`. Also please (as the above comment says) use paths instead of `File` objects...